### PR TITLE
Add exception handler middleware

### PR DESCRIPTION
**Pull request by samdark**
Should be useful for mapping domain exceptions to certain HTTP response codes.

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: src/Middleware/ExceptionResponder.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+use Yiisoft\Injector\Injector;
+
+class ExceptionResponder implements MiddlewareInterface
+{
+    private array $exceptionMap;
+    private ResponseFactoryInterface $responseFactory;
+    private Injector $injector;
+
+    /**
+     * @param array $exceptionMap
```
Comment (roxblnfk) [inline src/Middleware/ExceptionResponder.php:19]:
```suggestion
     * @param int[]|callable[] $exceptionMap Closure must return ResponseInterface
```

File: src/Middleware/ExceptionResponder.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<?php
+
```
Comment (roxblnfk) [inline src/Middleware/ExceptionResponder.php:2]:
```suggestion

declare(strict_types=1);

```

File: src/Middleware/ExceptionResponder.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+{
+    private array $exceptionMap;
+    private ResponseFactoryInterface $responseFactory;
+    private Injector $injector;
+
+    /**
+     * @param array $exceptionMap
+     * @psalm-param array{string, int|callable}
+     */
+    public function __construct(array $exceptionMap, ResponseFactoryInterface $responseFactory, Injector $injector)
```
Comment (roxblnfk) [inline src/Middleware/ExceptionResponder.php:22]: You can specify the ContainerInterface in the parameters and get the ResponseFactoryInterface and Injector already in the catch body

Comment (samdark) [reply src/Middleware/ExceptionResponder.php:24]: I don't see a profit in doing it. Makes code less clean.

Comment (roxblnfk) [reply src/Middleware/ExceptionResponder.php:24]: Mb. I just don't like that you pass the exception map along with the dependencies.

**Review by rustamwin — APPROVED** [review]
I like the idea.

**Comment by samdark** [comment]
This one can potentially become part of https://github.com/yiisoft/error-handler

**Pull request closed without merging**